src/main/java/com/google/devtools/build/lib/bazel/repository/cache/RepoContentsCache.java

```diff
@@ -159,9 +158,13 @@ public Path moveToCache(
     if (!entryDir.isDirectory()) {
       entryDir.delete();
     }
-    String counter = getNextCounterInDir(entryDir);
-    Path cacheRecordedInputsFile = entryDir.getChild(counter + RECORDED_INPUTS_SUFFIX);
-    Path cacheRepoDir = entryDir.getChild(counter);
+    // The name of the entry must be unique and remain so over time, even in case of data loss: If
+    // the name of an entry is ever reused, a Bazel output base that linked to the previous version
+    // of the entry would pick up the new version of the entry without invalidation as the marker
+    // file is not symlinked, but kept under that particular output base.
```

Comment:
there shouldn't be a marker file in the output base -- see RepoContentsCache.java:171